src/Model.jl

```diff
@@ -85,8 +88,8 @@ function Model(lattice::AbstractMatrix{T};
                spin_polarization=default_spin_polarization(magnetic_moments),
                symmetries=default_symmetries(lattice, atoms, magnetic_moments, terms, spin_polarization),
                ) where {T <: Real}
-    lattice = Mat3{T}(lattice)
-    temperature = T(temperature)
+    lattice = austrip.(Mat3{T}(lattice))
```

Comment:
I think you want to strip before converting to Mat3{T} here or it won't work

Reply:
It seems to work fine:
```julia
julia> m = [1.1 2.2 3.3; 4.4 5.5 6.6; 7.7 8.8 9.9]
3×3 Array{Float64,2}:
 1.1  2.2  3.3
 4.4  5.5  6.6
 7.7  8.8  9.9

julia> ms = Mat3(m)
3×3 SArray{Tuple{3,3},Float64,2,9} with indices SOneTo(3)×SOneTo(3):
 1.1  2.2  3.3
 4.4  5.5  6.6
 7.7  8.8  9.9

julia> msa = ms * 1u"angstrom"
3×3 SArray{Tuple{3,3},Quantity{Float64,𝐋,Unitful.FreeUnits{(Å,),𝐋,nothing}},2,9} with indices SOneTo(3)×SOneTo(3):
 1.1 Å  2.2 Å  3.3 Å
 4.4 Å  5.5 Å  6.6 Å
 7.7 Å  8.8 Å  9.9 Å

julia> austrip.(msa)
3×3 SArray{Tuple{3,3},Float64,2,9} with indices SOneTo(3)×SOneTo(3):
  2.0787    4.1574   6.2361
  8.31479  10.3935  12.4722
 14.5509   16.6296  18.7083
```

Reply:
```
julia> austrip.(Mat3{Float64}(randn(3,3)*1u"angstrom"))
ERROR: DimensionError:  and ha are not dimensionally compatible.
```

Reply:
Also I am not sure this constructor actually works as we intent. If `lattice` is a Array{Quantity} than `T` will be bound to `Quantity`, which is not `<: Real`, so this constructor anyway does not apply. I think I'd solve this by having another constructor `Model(lattice::AbstractArray{Q}; kwargs...) where {Q <: Quantity}` which does the step to strip away the units.